cpp/arcticdb/toolbox/library_tool.cpp

```diff
@@ -37,6 +37,22 @@ Segment LibraryTool::read_to_segment(const VariantKey& key) {
     return kv.segment();
 }
 
+std::optional<google::protobuf::Any> LibraryTool::read_metadata(const VariantKey& key){
+    auto store = std::make_shared<arcticdb::async::AsyncStore<util::SysClock>>(lib_, codec::default_lz4_codec(), encoding_version(lib_->config()));
```

Comment:
I'm just wondering why do we need a shared_ptr?

Reply:
Good point, I just copied it from another method which needed the shared_ptr. 
Instead I created a commit to use a `store_` property to be used by all methods, instead of creating it every time.

Reply:
Is there a reason to create a pointer at all?
1. Can we use const ref instead of a pointer?
2. If we can't use const ref I think `std::unique_ptr` is better.

Reply:
As discussed offline:

We keep using a `shared_ptr<AsyncStore>` because it is required to pass in as an argument to `delete_all_keys_of_type`.

Instead left a TODO comment to remove the `shared_ptr` when `delete_all_keys_of_type` changes its argument to a const ref.